playbooks/provision_nodes.yml

```diff
@@ -258,4 +262,144 @@
+      set_fact: connections={{ server_profile_templates.0.connectionSettings.connections | selectattr('name', 'equalto', ov_ansible_connection_name) | list }}
+    - fail:
+        msg: "The connection {{ ov_ansible_connection_name }} could not be found in template {{ ov_template }}"
+      when: connections | length < 1
+    - set_fact:
+        connection: "{{ connections | first }}"
+
+#    - name: Find the redundant connection that connects to the ansible network by its name
+#      set_fact: connections={{ server_profile_templates.0.connectionSettings.connections | selectattr('name', 'equalto', ov_ansible_redundant_connection_name) | list }}
+#    - fail:
+#        msg: "The connection {{ ov_ansible_redundant_connection_name }} could not be found in template {{ ov_template }}"
+#      when: connections | length < 1
+#    - set_fact:
+#        connection2: "{{ connections | first }}"
+
+    - name: Gather facts about a Server Profile by name
+      oneview_server_profile_facts:
+        hostname: "{{ oneview_config_hostname }}"
+        username: "{{ oneview_config_username }}"
+        password: "{{ oneview_config_password }}"
+        api_version: "{{ oneview_config_api_version }}"
+        name: "{{ inventory_hostname }}"
+      delegate_to: localhost
+
+    - name: Gather facts about server hardware
+      oneview_server_hardware_facts:
+        hostname: "{{ oneview_config_hostname }}"
+        username: "{{ oneview_config_username }}"
+        password: "{{ oneview_config_password }}"
+        api_version: "{{ oneview_config_api_version }}"
+        name: "{{ enclosure }}, bay {{ server_profiles[0].enclosureBay }}"
+      when: server_profiles | length > 0
+      delegate_to: localhost
+
+    #
+    # Don't try to change the server profile if the server is powered on
+    #
+    - set_fact: touch_profile=true
+    - set_fact: touch_profile='{% if server_hardwares[0].powerState != 'Off' %}false{% else %}true{% endif %}'
+      when: server_hardwares is defined and server_hardwares | length > 0
+    - debug: var=touch_profile
+
+    - name : "Provision Server"
+      delegate_to: localhost
+      when: touch_profile == true
+      oneview_server_profile:
+        hostname: "{{ oneview_config_hostname }}"
+        username: "{{ oneview_config_username }}"
+        password: "{{ oneview_config_password }}"
+        api_version: "{{ oneview_config_api_version }}"
+        state: "present"
+        data:
+          name: "{{ inventory_hostname }}"
+          serverProfileTemplateName: "{{ ov_template }}"
+          serverHardwareName: "{% if bay is defined %}{{ enclosure }}, bay {{ bay }}{% endif %}"
+          osDeploymentSettings:
+            osDeploymentPlanUri: "{{ server_profile_templates[0].osDeploymentSettings.osDeploymentPlanUri }}"
+            osCustomAttributes:
+             - name: Hostname
+               value: '{{ inventory_hostname }}'
+             - name: WindowsProductKey
```

Comment:
this should be set as not visible as I explained yesterday. There is not reason why the windows worker should be handled differently than the linux boxes

Reply:
I disagree.  The Windows Product Key is a private license key and should be stored in the encrypted vault like all other secret information.  I do agree that one goal is when deploying bare metal boxes they should be as similar as possible but there are some real differences between Windows and Linux systems that have to be taken into account when deploying.